vars/piperExecuteBin.groovy

```diff
@@ -31,7 +31,10 @@ void call(Map parameters = [:], stepName, metadataFile, List credentialInfo, fai
         def jenkinsUtils = parameters.jenkinsUtilsStub ?: new JenkinsUtils()
         stepParameters.remove('jenkinsUtilsStub')
 
-        new PiperGoUtils(this, utils).unstashPiperBin()
+        def piperGoUtils = parameters.piperGoUtils ?: new PiperGoUtils(this, utils)
```

Comment:
`parameters.piperGoUtils` are only set during tests, isn't it? Do we expect some cases where they are also provided in the free wild? If not that doesn't really change anything. Or would we like to impove some behaviour inside the tests?

Reply:
It is not for tesing ...

This makes e.g. an internal re-use possible.
We are actually using it like that to get rid of yet another copy of the same behavior.

By allowing to provide a custom `piperGoUtils` the download functionality is customized.